sse_client/event_test.go

```diff
@@ -0,0 +1,132 @@
+package sse_client
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"strings"
+)
+
+func TestParseEvent_shouldParseLongDataField(t *testing.T) {
+	t.Parallel()
+	// given
+	data := make([]byte, 10*1024*1024)
+	for i := range data {
+		data[i] = '-'
+	}
+	data = append(data, '\n')
+	message := append(b(`data:`), data...)
+	events := bufio.NewReader(bytes.NewReader(message))
+
+	// when
+	event, err := parseEvent(events)
+
+	// then
+	assert.EqualError(t, err, io.EOF.Error())
+	assert.Equal(t, string(data), string(event.Data))
+}
+
+func TestParseEvent_shouldReturnErrOnEOFWhenEventIsNotReady(t *testing.T) {
+	t.Parallel()
+	// given
+	events := bufio.NewReader(bytes.NewReader([]byte{}))
+
+	// when
+	_, err := parseEvent(events)
+
+	// then
+	assert.EqualError(t, err, "Unexpected EOF")
+}
+
+func TestParseEvent(t *testing.T) {
+	testCases := []struct {
```

Comment:
JSON does not support multi line strings. How this should be done? This Test cases are really small. What I will achieve with moving them to file?